src/walk.rs

```diff
@@ -289,6 +289,21 @@ pub fn scan(path_vec: &[PathBuf], pattern: Arc<Regex>, config: Arc<FdOptions>) {
                 }
             }
 
+            // Filter out unwanted modification times
+            if !config.modification_constraints.is_empty() {
+                if_chain!{
+                    if entry_path.is_file();
+                    if let Ok(metadata) = entry_path.metadata();
+                    if let Ok(modified) = metadata.modified();
+                    if config.modification_constraints.iter().all(|tf| tf.is_within(&modified));
```

Comment:
If we would invert the logic here by using `.iter().any(|tf| !tf.is_within(&modified)` we could profit from short-circuiting and also prevent the empty `if`/`then` clause.

Reply:
`all` also benefits from the same short-circuiting (as-soon as a predicate fails, `all` returns).
in this case, it allows to have all the `return WalkState::Continue` as the else branch, and thus put them in common.

Reply:
You are right, of course :+1: